test/suites/integration/httproute_method_match_test.go

```diff
@@ -89,40 +99,38 @@ var _ = Describe("HTTPRoute method matches", func() {
 
 		//get the pods of deployment1
 		pods := testFramework.GetPodsByDeploymentName(deployment1.Name, deployment1.Namespace)
-		cmd1 := fmt.Sprintf("curl -X GET %s", dnsName)
-		stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd1, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("test-get handler pod"))
-
-		cmd2 := fmt.Sprintf("curl -X POST %s", dnsName)
-		stdout, _, err = testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd2, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("test-post handler pod"))
-
-		invalidCmd := fmt.Sprintf("curl -X DELETE %s", dnsName)
-		stdout, _, err = testFramework.PodExec(pods[0].Namespace, pods[0].Name, invalidCmd, true)
-		Expect(err).To(BeNil())
-		Expect(stdout).To(ContainSubstring("Not Found"))
-
-		testFramework.ExpectDeleted(ctx,
-			gateway,
+
+		Eventually(func(g Gomega) {
+			cmd := fmt.Sprintf("curl -X GET %s", dnsName)
+			stdout, _, err := testFramework.PodExec(pods[0].Namespace, pods[0].Name, cmd, true)
+			g.Expect(err).To(BeNil())
+			g.Expect(stdout).To(ContainSubstring("test-get handler pod"))
+		}).WithTimeout(30 * time.Second).WithOffset(1).Should(Succeed())
```

Comment:
nit: eventually itself is assertion block, that can assert return value of the function inside, for simple tests like this we can drop a few things. Assertion on substring should be enough without checking error

```go
func curlPod(pod Pod, curlCmd string) (string, error) {
		stdout, _, err := testFramework.PodExec(pod.Namespace, pod.Name, curlCmd, true)
                return stdout, err
}

Eventually(func() string {
	out, _ := curlPod(pods[0], fmt.Sprintf("curl -X POST %s", dnsName))
	return out
}).Should(ContainSubstring("200 OK"))
```